### PR TITLE
History

### PR DESCRIPTION
**Pull request by mohammadkhizerkhan**
added history feature
refactored like and watchlater code

### CODE REVIEW

File: src/components/SingleVideo.jsx [hunk 1]
```diff
@@ -56,10 +58,11 @@ function SingleVideo() {
           </div>
         </div>
         <div class="single-video-btns flex-row-center">
-          {likeState.liked.some((like) => like._id === video._id) ? (
+          {console.log(like)}
```
Comment (Pratik1005) [inline src/components/SingleVideo.jsx:61]: remove console log if not required

File: src/pages/History.jsx [hunk 2]
```diff
@@ -1,11 +1,64 @@
...
+import { useAuth, useHistory } from "../context";
+import { deleteAllHistory } from "../services";
+import { useNavigate } from "react-router-dom";
 
 function History() {
-    return (
+  const { token } = useAuth();
+  const navigate = useNavigate();
+  const { history, setHistory } = useHistory();
+  console.log(history);
```
Comment (Pratik1005) [inline src/pages/History.jsx:13]: same here

**Review by Pratik1005 — APPROVED** [review]
History is working properly, good to merge

File: src/components/VideoCard.jsx [hunk 3]
```diff
@@ -19,17 +27,21 @@ function VideoCard({ video }) {
     likes,
     uploaded,
   } = video;
+  // console.log(like.some((item) => item._id === video._id))
```
Comment (shraddha-amber) [inline src/components/VideoCard.jsx:30]: remove comments

File: src/components/VideoCard.jsx [hunk 4]
```diff
@@ -19,17 +27,21 @@ function VideoCard({ video }) {
...
   return (
```
Comment (JowelTisso) [inline src/components/VideoCard.jsx:31]: * remove unused codes

File: src/context/LikedContext.jsx [hunk 5]
```diff
@@ -1,16 +1,13 @@
-import { createContext, useContext,useReducer} from "react";
-import { LikedReducer } from "../reducer";
+import { createContext, useContext} from "react";
+import { useState } from "react";
```
Comment (shraddha-amber) [inline src/context/LikedContext.jsx:2]: could combine useState import with previous import

**Review by JowelTisso — APPROVED** [review]
* Clean code
* Just remember to remove all the unused codes and console logs before raising PR

**Review by shraddha-amber — APPROVED** [review]
Clean and modular code
Functionality works properly

File: src/context/LikedContext.jsx (hunk 5 above)
Comment (mohammadkhizerkhan) [reply src/context/LikedContext.jsx:2]: sure

**Pull request merged**